includes/class-amp-reader-themes.php

```diff
@@ -0,0 +1,546 @@
+				$response = (object) $response;
+			}
+
+			$supported_themes = array_diff(
+				AMP_Core_Theme_Sanitizer::get_supported_themes(),
+				[ 'twentyten' ] // Excluded because not responsive.
+			);
+
+			$supported_themes_from_response = array_filter(
+				$response->themes,
+				static function( $theme ) use ( $supported_themes ) {
+					return in_array( $theme->slug, $supported_themes, true );
+				}
+			);
+
+			return $supported_themes_from_response;
+		}
+
+		$themes = self::get_default_raw_reader_themes();
+
+		return $themes;
+	}
+
+	/**
+	 * Prepares a single theme.
+	 *
+	 * @since 1.6.0
+	 *
+	 * @param array $theme Theme data from the wordpress.org themes API.
+	 * @return array Prepared theme array.
+	 */
+	public function prepare_theme( $theme ) {
+		$prepared_theme = [];
+		$theme_array    = (array) $theme;
+
+		$keys = [
+			'name',
+			'slug',
+			'preview_url',
+			'screenshot_url',
+			'homepage',
+			'description',
+			'requires',
+			'requires_php',
+			'download_link',
+		];
+
+		$prepared_theme = array_filter(
+			$theme_array,
+			function( $key ) use ( $keys ) {
+				return in_array( $key, $keys, true );
+			},
+			ARRAY_FILTER_USE_KEY
+		);
+
+		foreach ( $keys as $key ) {
+			if ( ! array_key_exists( $key, $prepared_theme ) ) {
+				$prepared_theme[ $key ] = '';
+			}
+		}
+
+		return $prepared_theme;
+	}
+
+	/**
+	 * Provides the current theme name.
+	 *
+	 * @return string|bool The theme name, or false if the theme has errors.
+	 */
+	private function get_current_theme_name() {
+		if ( is_null( $this->current_theme_name ) ) {
+			$current_theme = wp_get_theme();
+
+			$this->current_theme_name = $current_theme->exists() ? $current_theme->get( 'Name' ) : false;
+		}
+
+		return $this->current_theme_name;
+	}
+
+	/**
+	 * Returns whether a theme can be installed on the system.
+	 *
+	 * @since 1.6.0
+	 *
+	 * @param array $theme Theme data.
+	 * @return bool True if themes can be installed.
+	 */
+	public function can_install_theme( $theme ) {
+		// @todo Add support for installing non-default reader themes. Until that is done, themes that are provided via
+		// the amp_reader_themes filter will show on the reader themes screen but will need to be manually installed on
+		// the site.
+		$default_reader_theme_slugs = wp_list_pluck( $this->get_default_raw_reader_themes(), 'slug' );
+		if ( ! in_array( $theme['slug'], $default_reader_theme_slugs, true ) ) {
+			return false;
+		}
+
+		if ( is_null( $this->can_install_themes ) ) {
+			if ( ! class_exists( 'WP_Upgrader' ) ) {
+				require_once ABSPATH . 'wp-admin/includes/file.php';
+				require_once ABSPATH . 'wp-admin/includes/class-wp-upgrader.php';
+			}
+
+			$this->can_install_themes = true === ( new WP_Upgrader() )->fs_connect( [ get_theme_root() ] );
+		}
+
+		if ( ! $this->can_install_themes ) {
+			return false;
+		}
+
+		if ( ! empty( $theme['requires'] ) && ! version_compare( get_bloginfo( 'version' ), $theme['requires'], '>=' ) ) {
```

Comment:
No need to negate the result of `version_compare` in an additional computation if you can just reverse the condition to check.
```suggestion
		if ( ! empty( $theme['requires'] ) && version_compare( get_bloginfo( 'version' ), $theme['requires'], '<' ) ) {
```